### PR TITLE
update drm cursor matrix on set_transform

### PR DESCRIPTION
**Pull request by Ongy**
Updates the projection matrix for the cursor plane in the DRM backend,
when the output transform is changed, so new cursor are uploaded with
the correct transformation.

Testplan:
Get a setup that supports changing cursor transform during runtime, do it on DRM backend.
Before: Cursor transformation stays the way it was when the output was first set up (even new cursor)
After: Cursors are transformed properly

The *current* cursor set, when the transformation is applied may still be rotated the wrong way, since it's client owned and we do not have it anymore to update.

### CODE REVIEW

File: backend/drm/drm.c [hunk 1]
```diff
@@ -535,6 +535,17 @@ static bool drm_connector_set_mode(struct wlr_output *output,
...
 		enum wl_output_transform transform) {
 	output->transform = transform;
+
+	struct wlr_drm_connector *conn = (struct wlr_drm_connector *)output;
+	struct wlr_drm_crtc *crtc = conn->crtc;
+	if (!crtc) {
+		return;
+	}
+
+	if (crtc->cursor) {
```
[the remark this answers is not in the document]
Comment (emersion) [reply backend/drm/drm.c]: Could also just re-compute the matrix each time, pretty sure the overhead is negligible.

[the remark this answers is not in the document]
Comment (Ongy) [reply backend/drm/drm.c]: And not store it? Probably. I just did the least intrusive change I could think of

[the remark this answers is not in the document]
Comment (emersion) [reply backend/drm/drm.c]: Yeah, that would simplify the code and be consistent with the other rendering bits.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**